From dee5ebe3bb103dae4bc0c248083cc087535d9183 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:05:45 -0400
Subject: [usb] hub: usb_ext_cap_descriptor.bmAttributes is le32

Message-id: <1410887463-20674-274-git-send-email-dzickus@redhat.com>
Patchwork-id: 95228
O-Subject: [RHEL7 PATCH 273/591] usb: hub: usb_ext_cap_descriptor.bmAttributes is le32
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 23c058201fe36adf1a225281739b3ec31ec4e858
Author: Bjorn Mork <bjorn@mork.no>
Date:   Thu Feb 27 14:23:55 2014 +0100

    usb: hub: usb_ext_cap_descriptor.bmAttributes is le32

    Better check the correct bit on big endian systems too. Shuts
    up the following sparse __CHECK_ENDIAN__ warning:

      .../hub.c:3965:32: warning: restricted __le32 degrades to integer

    Signed-off-by: Bjorn Mork <bjorn@mork.no>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/core/hub.c b/drivers/usb/core/hub.c
index 9cf1114..c5ec48d 100644
--- a/drivers/usb/core/hub.c
+++ b/drivers/usb/core/hub.c
@@ -3963,7 +3963,7 @@ static void hub_set_initial_usb2_lpm_policy(struct usb_device *udev)
  connect_type = usb_get_hub_port_connect_type(udev->parent,
    udev->portnum);
 
- if ((udev->bos->ext_cap->bmAttributes & USB_BESL_SUPPORT) ||
+ if ((udev->bos->ext_cap->bmAttributes & cpu_to_le32(USB_BESL_SUPPORT)) ||
    connect_type == USB_PORT_CONNECT_TYPE_HARD_WIRED) {
   udev->usb2_hw_lpm_allowed = 1;
   usb_set_usb2_hardware_lpm(udev, 1);
-- 
1.7.1